packager/media/formats/mp2t/ts_muxer.cc

```diff
@@ -34,12 +30,16 @@ Status TsMuxer::Finalize() {
 
 Status TsMuxer::AddSample(size_t stream_id, const MediaSample& sample) {
   DCHECK_EQ(stream_id, 0u);
+  sample_duration_ = sample.duration();
```

Comment:
I think you can call OnSampleDurationReady here, i.e.

```c++
if (sample_duration_ == 0) {
  sample_duration_ = sample.duration() * kTsTimescale / streams().front()->time_scale();
  if (muxer_listener()) 
    muxer_listener()->OnSampleDurationReady(sample_duration_);
} 
```

`if (sample_duration_ == 0)` is needed to avoid issuing event multiple times, which is needed only once.

The below code is needed to re-scale the timestamp to output timescale.

```c++
sample.duration() * kTsTimescale / streams().front()->time_scale()
```

Reply:
Changed, it seems to work properly with a frame rate of 24. Currently working on unit tests

Reply:
Looks like there isn't a unit-test. I think it is fine to just update the end to end test.